lib/types/uuid.js

```diff
@@ -2,126 +2,150 @@
-    return new Uuid(
-        utils.allocBufferFromString(value.replace(/-/g, ""), "hex"),
-    );
-};
 
-/**
- * Creates a new random (version 4) Uuid.
- * @param {function} [callback] Optional callback to be invoked with the error as first parameter and the created Uuid as
- * second parameter.
- * @returns {Uuid}
- */
-Uuid.random = function (callback) {
-    if (callback) {
-        getRandomBytes(function (err, buffer) {
-            if (err) {
-                return callback(err);
+    /**
+     * Creates a new random (version 4) Uuid.
+     * @param {function} [callback] Optional callback to be invoked with the error as first parameter and the created Uuid as
+     * second parameter.
+     * @returns {Uuid}
+     */
+    static random(callback) {
+        if (callback) {
+            getRandomBytes(function (err, buffer) {
+                if (err) {
+                    return callback(err);
+                }
+                return callback(null, createUuidFromBuffer(buffer));
+            });
+        } else {
+            const buffer = getRandomBytes();
+            return createUuidFromBuffer(buffer);
+        }
+    }
+
+    /**
+     * Gets the bytes representation of a Uuid
+     * @returns {Buffer}
+     */
+    getBuffer() {
+        return this.internal.getBuffer();
+    }
+
+    /**
+     * Compares this object to the specified object.
+     * The result is true if and only if the argument is not null, is a UUID object, and contains the same value, bit for bit, as this UUID.
+     * @param {Uuid} other The other value to test for equality.
+     */
+    equals(other) {
+        if (!(other instanceof Uuid)) {
+            return false;
+        }
+        const thisBuffer = this.buffer;
+        const otherBuffer = other.getBuffer();
+        for (let i = 0; i < 16; i++) {
+            if (thisBuffer[i] !== otherBuffer[i]) {
+                return false;
             }
-            return callback(null, createUuidFromBuffer(buffer));
-        });
-    } else {
-        const buffer = getRandomBytes();
-        return createUuidFromBuffer(buffer);
+        }
+        return true;
     }
-};
 
-/**
- * Gets the bytes representation of a Uuid
- * @returns {Buffer}
- */
-Uuid.prototype.getBuffer = function () {
-    return this.buffer;
-};
-/**
- * Compares this object to the specified object.
- * The result is true if and only if the argument is not null, is a UUID object, and contains the same value, bit for bit, as this UUID.
- * @param {Uuid} other The other value to test for equality.
- */
-Uuid.prototype.equals = function (other) {
-    return other instanceof Uuid && this.buffer.equals(other.buffer);
-};
+    /**
+     * Returns a string representation of the value of this Uuid instance.
+     * 32 hex separated by hyphens, in the form of 00000000-0000-0000-0000-000000000000.
+     * @returns {String}
+     */
+    toString() {
+        // 32 hex representation of the Buffer
+        const hexValue = getHex(this);
```

Comment:
Helper function with just one call. Is it required to be a helper function then?